apps/api-documenter/src/markdown/CustomMarkdownEmitter.ts

```diff
@@ -0,0 +1,187 @@
+          if (row.cells.length > columnCount) {
+            columnCount = row.cells.length;
+          }
+        }
+
+        // write the table header (which is required by Markdown)
+        writer.write('| ');
+        for (let i: number = 0; i < columnCount; ++i) {
+          writer.write(' ');
+          if (docTable.header) {
+            const cell: DocTableCell | undefined = docTable.header.cells[i];
+            if (cell) {
+              this.writeNode(cell.content, context);
+            }
+          }
+          writer.write(' |');
+        }
+        writer.writeLine();
+
+        // write the divider
+        writer.write('| ');
+        for (let i: number = 0; i < columnCount; ++i) {
+          writer.write(' --- |');
+        }
+        writer.writeLine();
+
+        for (const row of docTable.rows) {
+          writer.write('| ');
+          for (const cell of row.cells) {
+            writer.write(' ');
+            this.writeNode(cell.content, context);
+            writer.write(' |');
+          }
+          writer.writeLine();
+        }
+        writer.writeLine();
+
+        context.insideTable = false;
+
+        break;
+      }
+      case CustomDocNodeKind.EmphasisSpan: {
+        const docEmphasisSpan: DocEmphasisSpan = docNode as DocEmphasisSpan;
+        const oldBold: boolean = context.boldRequested;
+        const oldItalic: boolean = context.italicRequested;
+        context.boldRequested = docEmphasisSpan.bold;
+        context.italicRequested = docEmphasisSpan.italic;
+        this.writeNodes(docEmphasisSpan.nodes, context);
+        context.boldRequested = oldBold;
+        context.italicRequested = oldItalic;
+        break;
+      }
+      default:
+        super.writeNode(docNode, context);
+    }
+  }
+
+  /** @override */
+  protected writeLinkTagWithCodeDestination(docLinkTag: DocLinkTag,
+    context: IMarkdownEmitterContext<ICustomMarkdownEmitterOptions>): void {
+
+    const options: ICustomMarkdownEmitterOptions = context.options;
+
+    const result: IResolveDeclarationReferenceResult
+      = this._apiModel.resolveDeclarationReference(docLinkTag.codeDestination!, options.contextApiItem);
+
+    if (result.resolvedApiItem) {
+      const filename: string | undefined = options.onGetFilenameForApiItem(result.resolvedApiItem);
+
+      if (filename) {
+        let linkText: string = docLinkTag.linkText || '';
+        if (linkText.length === 0) {
+
+          // Generate a name such as Namespace1.Namespace2.MyClass.myMethod()
+          linkText = result.resolvedApiItem.getScopedNameWithinPackage();
+        }
+        if (linkText.length > 0) {
+          const encodedLinkText: string = this.getEscapedText(linkText.replace(/\s+/g, ' '));
+
+          context.writer.write('[');
+          context.writer.write(encodedLinkText);
+          context.writer.write(`](${filename!})`);
+        } else {
+          console.log(colors.red('WARNING: Unable to determine link text'));
```

Comment:
>console.log(colors.red('WARNING: Unable to determine link text')); [](start = 10, length = 66)

It might be useful to output the link / link text which resulted in the above error.

Reply:
Yeah, we should probably fix that in this PR.  Ideally we'd also give a filename or line number, but `CustomMarkdownEmitter` doesn't easily have access to that information, so it's probably out of scope for this PR.

---
In reply to: [236550552](https://github.com/Microsoft/web-build-tools/pull/952#discussion_r236550552) [](ancestors = 236550552)